### PR TITLE
5869 android query rendered features

### PR DESCRIPTION
**Pull request by ivovandongen**
Ready for testing

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java [hunk 1]
```diff
@@ -1722,6 +1725,33 @@ public void snapshot(@NonNull SnapshotReadyCallback callback) {
...
+    /**
+     * Queries the map for rendered features
+     *
+     * @param coordinates the point to query
+     * @param layerIds    optionally - only query these layers
+     * @return the list of feature
+     */
+    @UiThread
+    @NonNull
+    public List<Feature> queryRenderedFeatures(@NonNull PointF coordinates, @Nullable String... layerIds) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: varargs is great for this use-case!

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/NativeMapView.java [hunk 2]
```diff
@@ -42,6 +45,8 @@
     // Used for callbacks
     private MapView mMapView;
 
+    private final float pixelRatio;
+
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/NativeMapView.java]: maybe we should move all density related calls to NativeMapView in favor of MapView

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/feature/QueryRenderedFeaturesBoxHighlightActivity.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void setupActionBar() {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/feature/QueryRenderedFeaturesBoxHighlightActivity.java]: composition this into utility method? reuse in other activities?

File: platform/android/src/jni/local_object.hpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <jni/jni.hpp>
+
```
Comment (tobrun) [inline platform/android/src/jni/local_object.hpp]: could be part of jni.hpp library?

**Comment by tobrun** [comment]
Added some nit remarks, but LGTM 🚢

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/NativeMapView.java (hunk 2 above)
Comment (ivovandongen) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/NativeMapView.java]: Considered that, but the native map view already has to know about density on both sides anyway.

**Comment by ivovandongen** [comment]
cc @mapbox/android

**Comment by zugaldia** [comment]
@ivovandongen 💥

**Pull request merged**